### PR TITLE
[single-cluster/aws] make `self-signed` work

### PR DESCRIPTION
**Pull request by Pothulapati**
## Description

This PR adds a new `post_bootstrap_user_data` variable to the EKS terraform modules that changes the containerd config to accept new self signed SSL Certificates, and restarting `containerd` to make the same work.

This PR also changes the `self-signed` of infra tests to upload the CA certificate into gcs for us to download and use.

Signed-off-by: Tarun Pothulapati <tarun@gitpod.io>

## Related Issue(s)

Fixes https://github.com/gitpod-io/gitpod/issues/12979

## How to test

Run
```
werft run github -f -s .werft/installer-tests.ts -j .werft/eks-installer-tests.yaml -a debug=true -a selfSigned=true -a skipTests=true -a preview=true -a domain=tests.doptig.com
```

or use the `self-signed` instance at `https://f0761-aws.tests.doptig.com` while using the certificate from https://werft.gitpod-dev.com/job/gitpod-custom-tar-eks-selfsigned.6/results

## Release Notes

```release-note
[single-cluster/aws] Make `self-signed` work
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: .werft/eks-installer-tests.yaml [hunk 1]
```diff
@@ -89,6 +89,8 @@ pod:
       value: "/mnt/secrets/sh-playground-sa-perm/sh-sa.json"
     - name: TF_VAR_dns_sa_creds
       value: "/mnt/secrets/sh-playground-dns-perm/sh-dns-sa.json"
+    - name: TF_VAR_sa_creds
```
Comment (Pothulapati) [inline .werft/eks-installer-tests.yaml:92]: This is an unrelated change required to re-use the same environment while running tests! Context: https://gitpod.slack.com/archives/C01KLC56NP7/p1664268857397169?thread_ts=1664261197.651119&cid=C01KLC56NP7

File: install/tests/Makefile [hunk 2]
```diff
@@ -314,6 +314,9 @@ self-signed-config:
 	envsubst < ./manifests/kots-config-self-signed.yaml > tmp_2_config.yml
 	yq m -i tmp_config.yml tmp_2_config.yml
 
+	# upload the Custom CA Cert into tf-state
```
Comment (nandajavarma) [inline install/tests/Makefile:317]: Can you also add commands to delete this file upon cleanup? It can be i[n this target ](https://github.com/gitpod-io/gitpod/pull/13305/files#diff-bd7b7f3a397df93111a6b6e2a4284b01bd57e04474bc9f91f1b02ee1960399a2L448) if you would like.

File: install/infra/modules/eks/kubernetes.tf [hunk 3]
```diff
@@ -103,6 +103,18 @@ module "eks" {
     enable_bootstrap_user_data = true
     vpc_security_group_ids     = [aws_security_group.nodes.id]
     ebs_optimized              = true
+
```
Comment (nandajavarma) [inline install/infra/modules/eks/kubernetes.tf:106]: <3

**Review by nandajavarma — APPROVED** [review]
Just a small cleanup command is necessary. This is a nit but can go long way in keeping the cloud storage clean.

/hold merge at your discretion

Would you also consider adding a doc for this to the [internal notion page](https://www.notion.so/gitpod/Self-hosted-automated-preview-and-test-pipelines-fdc5a202e67b4197aa00d93b98d57927). The page doesn't have the most easiest structure right now but having it there would be great!

File: install/tests/Makefile (hunk 2 above)
Comment (Pothulapati) [reply install/tests/Makefile:317]: I've added it to `destroy-kubeconfig`, Do you think we should do it separately? 🤔
